checkpoint/recovery.go

```diff
@@ -89,11 +99,23 @@ func copyToLocalFile(
 		logger.Info("old recovery data backed up", log.ZContext(ctx), zap.String("dir", bdir))
 	}
 	dst := RecoveryFilename(dataDir, filepath.Base(parsed.String()), restore)
-	if err = httpToLocalFile(ctx, parsed, fs, dst); err != nil {
-		return "", err
+	for range retryCount + 1 {
```

Comment:
We already depend on and use https://github.com/hashicorp/go-retryablehttp. How about using it instead of implementing retries manually?

Reply:
Makes sense, I'll update the PR

Reply:
Fixed